libraries/helper.rb

```diff
@@ -46,22 +46,44 @@ def omnibus_env
       node.run_state[:omnibus_env] ||= Hash.new { |hash, key| hash[key] = [] } # ~FC001
     end
 
-    def omnibus_toolchain_enabled?
-      # We don't have an omnibus toolchain for windows yet.
-      !windows?
+    def toolchain_install_dir
+      if windows?
+        windows_safe_path_join(ENV['SYSTEMDRIVE'], 'opscode', node['omnibus']['toolchain_name'])
+      else
+        "/opt/#{node['omnibus']['toolchain_name']}"
+      end
     end
 
     def build_user_shell
-      if omnibus_toolchain_enabled?
-        "/opt/#{node['omnibus']['toolchain_name']}/bin/bash"
+      if windows?
+        windows_safe_path_join(toolchain_install_dir, 'embedded', 'bin', 'usr', 'bin', 'bash')
       else
-        '/usr/local/bin/bash'
+        ::File.join(toolchain_install_dir, 'bin', 'bash')
       end
     end
 
     def windows_arch_i386?
       windows? && (i386? || (node.name =~ /i386/))
     end
+
+    def mingw_toolchain_name
+      windows_arch_i386? ? 'mingw32' : 'mingw64'
+    end
+
+    def msys2_path(path)
+      path.sub(%r{^([A-Za-z]):/}, '/\\1/')
```

Comment:
So I copied this from @ksubrama's stuff. The end result _should_ be `C:/foo/bar` should be turned into `/c/foo/bar`. Basically, the same result as running `cygpath`.